### PR TITLE
Gradle distribution filename

### PR DESCRIPTION
**Pull request by scphantm**
This adds the ability to change the file name pattern for the gradle distribution for corp users that rebranded their own versions of gradle.

### CODE REVIEW

**Comment by ysb33r** [comment]
I have not had a chance to look at this yet, but will do before end of this year.

File: src/main/groovy/org/ysb33r/gradle/gradletest/GradleTest.groovy [hunk 1]
```diff
@@ -146,6 +146,19 @@ class GradleTest extends Test {
         baseDistributionUri
     }
 
+    String getGradleDistributionFilenamePattern() {
```
Comment (scphantm) [inline src/main/groovy/org/ysb33r/gradle/gradletest/GradleTest.groovy:150]: Adding the filename pattern to the GradleTest task

File: src/main/groovy/org/ysb33r/gradle/gradletest/GradleTest.groovy [hunk 2]
```diff
@@ -264,6 +277,7 @@ class GradleTest extends Test {
     private List<Object> arguments = [/*'--no-daemon',*/ '--full-stacktrace', '--info'] as List<Object>
     private List<Object> versions = []
     private URI baseDistributionUri
+    private String distributionFilenamePattern = '/gradle-@version@-bin.zip'
```
Comment (scphantm) [inline src/main/groovy/org/ysb33r/gradle/gradletest/GradleTest.groovy:294]: Setting the default of the filename pattern.

File: src/main/groovy/org/ysb33r/gradle/gradletest/TestGenerator.groovy [hunk 3]
```diff
@@ -446,7 +451,8 @@ class TestGenerator extends DefaultTask {
                 SOURCEDIR : src,
                 FAILMODE : willFail,
                 CHECK_WARNINGS : deprecation,
-                DELETE_SCRIPT : deleteScript
+                DELETE_SCRIPT : deleteScript,
+                DIST_FILENAME_PATTERN : gradleFilenamePattern
```
Comment (scphantm) [inline src/main/groovy/org/ysb33r/gradle/gradletest/TestGenerator.groovy:487]: Passing the filename pattern to the spec template file

File: src/main/resources/org/ysb33r/gradletest/GradleTestTemplate.groovy.template [hunk 4]
```diff
@@ -123,6 +123,17 @@ class ${TESTNAME.replaceAll(/\W/,'_')}CompatibilitySpec extends Specification {
         }
     }
 
+    def getDistributionFilename(String version){
```
Comment (scphantm) [inline src/main/resources/org/ysb33r/gradletest/GradleTestTemplate.groovy.template]: New code in the template file that does the actual version replace for the filename pattern and returns the final file name

**Comment by scphantm** [comment]
I added the comments to this one.  like i said, without thinking about it i merged 88 into 89, forgetting that it muddied up the MR.  i commented on the highlights.  the commit to really look at is 

https://github.com/ysb33r/gradleTest/pull/89/commits/c77a07db61de01a55a3a1c597567f58e2f7374ef

**Comment by ysb33r** [comment]
Changes look good on this one. Do you want me to merge it?

**Comment by scphantm** [comment]
merge in 88 first, i found a bunch of bugs with it i have been fixing the past few days.  this one was branched off 88 so its easier for it to go in first.

**Pull request merged**